test/test-client-server.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -e
+
+sock="/tmp/vfio-user.sock"
+rm -f ${sock}*
+../samples/server -v ${sock} &
+while [ ! -S ${sock} ]; do
+	sleep 0.1
+done
+../samples/client ${sock} &
+wait $!
```

Comment:
shouldn't this just be:

wait

Reply:
The problem with a simple `wait` is that it return the exit code of the last child that exited, so any other children that failed won't cause `wait` to fail, so the CI will succeed. In my tests it was usually the client that was failing. I'll try to find a way to make the whole script fail even if one child fails.

Reply:
fixed, take another look

Reply:
why do we even need the wait $! ? Can't we run the client synchronously?

Reply:
That is a very good point!

Reply:
Fixed, please take another look.